src/main/java/frc/robot/leds/LedStrip.java

```diff
@@ -0,0 +1,192 @@
+     */
+    public void setLength(int length) {
+        ledBuffer = new AddressableLEDBuffer(length);
+        ledStrip.setLength(length);
+    }
+
+    /**
+     * Sets the primary color of the strip.
+     *
+     * @param color Primary color.
+     */
+    public void setPrimary(Color color) {
+        primary = color;
+    }
+
+    /**
+     * Sets the secondary color of the strip.
+     *
+     * @param color Secondary color.
+     */
+    public void setSecondary(Color color) {
+        secondary = color;
+    }
+
+    /**
+     * Switch mode to solid color.
+     */
+    public void solid() {
+        mode = Mode.SOLID;
+    }
+
+    /**
+     * Switch mode to blink.
+     * (Should use setBlinkTime before this)
+     */
+    public void blink() {
+        mode = Mode.BLINK;
+    }
+
+    /**
+     * Switch mode to fade.
+     * (Should use setFadeDuration before this)
+     */
+    public void fade() {
+        mode = Mode.FADE;
+    }
+
+    /**
+     * Switch mode to rainbow.
+     */
+    public void rainbow() {
+        mode = Mode.RAINBOW;
+    }
+
+    /**
+     * Sets the led blink time.
+     *
+     * @param blinkTime Blink time in seconds
+     */
+    public void setBlinkTime(double blinkTime) {
+        this.blinkTime = blinkTime;
+    }
+
+    /**
+     * Sets the duration of the fade effect.
+     *
+     * @param duration Fade duration in seconds.
+     */
+    public void setFadeDuration(double duration) {
+        fadeDuration = duration;
+    }
+
+    /**
+     * Sets the led strip to a solid color
+     *
+     * @param color Color to set the strip to.
+     */
+    private void setSolidColor(Color color) {
+        var appliedColor = new edu.wpi.first.wpilibj.util.Color(color.getRed(), color.getGreen(), color.getBlue());
+        for (int i = 0; i < ledBuffer.getLength(); i++) {
+            ledBuffer.setLED(i, appliedColor);
+        }
+        ledStrip.setData(ledBuffer);
+    }
+
+    /**
+     * Interpolates between two colors.
+     *
+     * @param initial Initial color.
+     * @param goal    Final Color.
+     * @return
+     */
+    private Translation3d colorInterpolation(Color initial, Color goal) {
+        var initialHSB = Color.RGBtoHSB(initial.getRed(), initial.getGreen(), initial.getBlue(), new float[3]);
+        var goalHSB = Color.RGBtoHSB(goal.getRed(), goal.getGreen(), goal.getBlue(), new float[3]);
+
+        Translation3d initialPoint = new Translation3d(initialHSB[0], initialHSB[1], initialHSB[2]);
+        Translation3d goalPoint = new Translation3d(goalHSB[0], goalHSB[1], goalHSB[2]);
+
+        double d = initialPoint.getDistance(goalPoint) / fadeDuration;
+        double t = d / (initialPoint.minus(goalPoint)).getNorm();
+
+        solution = initialPoint.interpolate(goalPoint, t);
+        return solution;
+    }
+
+
+    /**
+     * Sets the led strip to a rainbow pattern.
+     */
+    private void setRainbow() {
+        for (int i = 0; i < ledBuffer.getLength(); i++) {
```

Comment:
`ledBuffer.getLength()` is used in a few places, it should be stored as a class member.